Terraform crash parsing a big integer during the upgrade to 0.12

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://github.com/terraform-providers .
-->

### Terraform Version
<!---
Run `terraform -v` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
v0.12.10
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file. For security, you can also encrypt the files using our GPG public key.
-->

```hcl
variable "axis_high_watermark" {
  default = 17976931348623157000
}

variable "axis_low_watermark" {
  default  = -17976931348623157000
}

variable "axis_max_value" {
  default = 17976931348623157000
}

variable "axis_min_value" {
  default = -17976931348623157000
}
```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.
-->

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->
```
2019/10/22 01:37:28 [INFO] Terraform version: 0.12.10  
2019/10/22 01:37:28 [INFO] Go runtime version: go1.12.9
2019/10/22 01:37:28 [INFO] CLI args: []string{"/usr/bin/terraform12", "0.12upgrade"}
2019/10/22 01:37:28 [DEBUG] Attempting to open CLI config file: /home/igor_mb/.terraformrc
2019/10/22 01:37:28 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019/10/22 01:37:28 [INFO] CLI command args: []string{"0.12upgrade"}
2019/10/22 01:37:28 [DEBUG] command: asking for input: "Would you like to upgrade the module in the current directory?"
2019/10/22 01:37:29 [DEBUG] checking for provider in "."
2019/10/22 01:37:29 [DEBUG] checking for provider in "/usr/bin"
2019/10/22 01:37:29 [DEBUG] checking for provider in ".terraform/plugins/linux_amd64"
2019/10/22 01:37:29 [DEBUG] found provider "terraform-provider-signalfx_v4.9.1_x4"
2019/10/22 01:37:29 [DEBUG] found valid plugin: "signalfx", "4.9.1", "/home/igor_mb/workspace/terraform-sda/signalfx/dashboards/.terraform/plugins/linux_amd64/terraform-provider-signalfx_v4.9.1_x4"
2019/10/22 01:37:29 [DEBUG] checking for provisioner in "."
2019/10/22 01:37:29 [DEBUG] checking for provisioner in "/usr/bin"
2019/10/22 01:37:29 [DEBUG] checking for provisioner in ".terraform/plugins/linux_amd64"
2019/10/22 01:37:29 [TRACE] configupgrade: Analyzing "prvider.tf"
2019/10/22 01:37:29 [TRACE] configupgrade: Analyzing "variables.tf"
2019/10/22 01:37:29 [TRACE] configupgrade: Analyzing "dashboard-groups.tf"
2019/10/22 01:37:29 [TRACE] Resource block for signalfx_dashboard_group.dashboardgroup_sda requires provider "signalfx"
2019/10/22 01:37:29 [TRACE] configupgrade: Analyzing "dashboards-kafka.tf"
2019/10/22 01:37:29 [ERROR] Failed to parse "dashboards-kafka.tf": At 25:13: unexpected token while parsing list: IDENT
2019/10/22 01:37:29 [TRACE] configupgrade: Analyzing "provider.tf"
2019/10/22 01:37:29 [TRACE] Provider block requires provider "signalfx"
2019/10/22 01:37:30 [TRACE] Fetching schema from provider "signalfx"
2019-10-22T01:37:30.121+0300 [INFO]  plugin: configuring client automatic mTLS
2019-10-22T01:37:30.153+0300 [DEBUG] plugin: starting plugin: path=/home/igor_mb/workspace/terraform-sda/signalfx/dashboards/.terraform/plugins/linux_amd64/terraform-provider-signalfx_v4.9.1_x4 args=[/home/igor_mb/workspace/terraform-sda/signalfx/dashboards/.terraform/plugins/linux_amd64/terraform-provider-signalfx_v4.9.1_x4]
2019-10-22T01:37:30.153+0300 [DEBUG] plugin: plugin started: path=/home/igor_mb/workspace/terraform-sda/signalfx/dashboards/.terraform/plugins/linux_amd64/terraform-provider-signalfx_v4.9.1_x4 pid=8039
2019-10-22T01:37:30.153+0300 [DEBUG] plugin: waiting for RPC address: path=/home/igor_mb/workspace/terraform-sda/signalfx/dashboards/.terraform/plugins/linux_amd64/terraform-provider-signalfx_v4.9.1_x4
2019-10-22T01:37:30.160+0300 [INFO]  plugin.terraform-provider-signalfx_v4.9.1_x4: configuring server automatic mTLS: timestamp=2019-10-22T01:37:30.159+0300
2019-10-22T01:37:30.192+0300 [DEBUG] plugin.terraform-provider-signalfx_v4.9.1_x4: plugin address: address=/tmp/plugin914104273 network=unix timestamp=2019-10-22T01:37:30.192+0300
2019-10-22T01:37:30.192+0300 [DEBUG] plugin: using plugin: version=5
2019/10/22 01:37:30 [TRACE] GRPCProvider: GetSchema
2019/10/22 01:37:30 [TRACE] Fetching schema from provisioner "chef"
2019-10-22T01:37:30.257+0300 [DEBUG] plugin: starting plugin: path=/usr/bin/terraform12 args=[/usr/bin/terraform12, internal-plugin, provisioner, chef]
2019-10-22T01:37:30.257+0300 [DEBUG] plugin: plugin started: path=/usr/bin/terraform12 pid=8053
2019-10-22T01:37:30.257+0300 [DEBUG] plugin: waiting for RPC address: path=/usr/bin/terraform12
2019-10-22T01:37:30.267+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Terraform version: 0.12.10  
2019-10-22T01:37:30.267+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Go runtime version: go1.12.9
2019-10-22T01:37:30.267+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI args: []string{"/usr/bin/terraform12", "internal-plugin", "provisioner", "chef"}
2019-10-22T01:37:30.267+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] Attempting to open CLI config file: /home/igor_mb/.terraformrc
2019-10-22T01:37:30.267+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019-10-22T01:37:30.267+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI command args: []string{"internal-plugin", "provisioner", "chef"}
2019-10-22T01:37:30.268+0300 [DEBUG] plugin.terraform12: chef-provisioner (internal) 2019/10/22 01:37:30 [INFO] Starting provisioner plugin chef
2019-10-22T01:37:30.268+0300 [DEBUG] plugin.terraform12: plugin address: address=/tmp/plugin723792245 network=unix timestamp=2019-10-22T01:37:30.268+0300
2019-10-22T01:37:30.268+0300 [DEBUG] plugin: using plugin: version=5
2019/10/22 01:37:30 [TRACE] Fetching schema from provisioner "file"
2019-10-22T01:37:30.270+0300 [DEBUG] plugin: starting plugin: path=/usr/bin/terraform12 args=[/usr/bin/terraform12, internal-plugin, provisioner, file]
2019-10-22T01:37:30.270+0300 [DEBUG] plugin: plugin started: path=/usr/bin/terraform12 pid=8064
2019-10-22T01:37:30.270+0300 [DEBUG] plugin: waiting for RPC address: path=/usr/bin/terraform12
2019-10-22T01:37:30.279+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Terraform version: 0.12.10  
2019-10-22T01:37:30.279+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Go runtime version: go1.12.9
2019-10-22T01:37:30.279+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI args: []string{"/usr/bin/terraform12", "internal-plugin", "provisioner", "file"}
2019-10-22T01:37:30.279+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] Attempting to open CLI config file: /home/igor_mb/.terraformrc
2019-10-22T01:37:30.279+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019-10-22T01:37:30.279+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI command args: []string{"internal-plugin", "provisioner", "file"}
2019-10-22T01:37:30.279+0300 [DEBUG] plugin.terraform12: file-provisioner (internal) 2019/10/22 01:37:30 [INFO] Starting provisioner plugin file
2019-10-22T01:37:30.279+0300 [DEBUG] plugin.terraform12: plugin address: address=/tmp/plugin017219757 network=unix timestamp=2019-10-22T01:37:30.279+0300
2019-10-22T01:37:30.279+0300 [DEBUG] plugin: using plugin: version=5
2019/10/22 01:37:30 [TRACE] Fetching schema from provisioner "habitat"
2019-10-22T01:37:30.281+0300 [DEBUG] plugin: starting plugin: path=/usr/bin/terraform12 args=[/usr/bin/terraform12, internal-plugin, provisioner, habitat]
2019-10-22T01:37:30.282+0300 [DEBUG] plugin: plugin started: path=/usr/bin/terraform12 pid=8073
2019-10-22T01:37:30.282+0300 [DEBUG] plugin: waiting for RPC address: path=/usr/bin/terraform12
2019-10-22T01:37:30.291+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Terraform version: 0.12.10  
2019-10-22T01:37:30.291+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Go runtime version: go1.12.9
2019-10-22T01:37:30.291+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI args: []string{"/usr/bin/terraform12", "internal-plugin", "provisioner", "habitat"}
2019-10-22T01:37:30.291+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] Attempting to open CLI config file: /home/igor_mb/.terraformrc
2019-10-22T01:37:30.291+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019-10-22T01:37:30.292+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI command args: []string{"internal-plugin", "provisioner", "habitat"}
2019-10-22T01:37:30.292+0300 [DEBUG] plugin.terraform12: habitat-provisioner (internal) 2019/10/22 01:37:30 [INFO] Starting provisioner plugin habitat
2019-10-22T01:37:30.292+0300 [DEBUG] plugin.terraform12: plugin address: address=/tmp/plugin199394191 network=unix timestamp=2019-10-22T01:37:30.292+0300
2019-10-22T01:37:30.292+0300 [DEBUG] plugin: using plugin: version=5
2019/10/22 01:37:30 [TRACE] Fetching schema from provisioner "local-exec"
2019-10-22T01:37:30.293+0300 [DEBUG] plugin: starting plugin: path=/usr/bin/terraform12 args=[/usr/bin/terraform12, internal-plugin, provisioner, local-exec]
2019-10-22T01:37:30.293+0300 [DEBUG] plugin: plugin started: path=/usr/bin/terraform12 pid=8083
2019-10-22T01:37:30.293+0300 [DEBUG] plugin: waiting for RPC address: path=/usr/bin/terraform12
2019-10-22T01:37:30.303+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Terraform version: 0.12.10  
2019-10-22T01:37:30.303+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Go runtime version: go1.12.9
2019-10-22T01:37:30.303+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI args: []string{"/usr/bin/terraform12", "internal-plugin", "provisioner", "local-exec"}
2019-10-22T01:37:30.303+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] Attempting to open CLI config file: /home/igor_mb/.terraformrc
2019-10-22T01:37:30.303+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019-10-22T01:37:30.303+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI command args: []string{"internal-plugin", "provisioner", "local-exec"}
2019-10-22T01:37:30.303+0300 [DEBUG] plugin.terraform12: local-exec-provisioner (internal) 2019/10/22 01:37:30 [INFO] Starting provisioner plugin local-exec
2019-10-22T01:37:30.303+0300 [DEBUG] plugin.terraform12: plugin address: address=/tmp/plugin557087575 network=unix timestamp=2019-10-22T01:37:30.303+0300
2019-10-22T01:37:30.304+0300 [DEBUG] plugin: using plugin: version=5
2019/10/22 01:37:30 [TRACE] Fetching schema from provisioner "puppet"
2019-10-22T01:37:30.304+0300 [DEBUG] plugin: starting plugin: path=/usr/bin/terraform12 args=[/usr/bin/terraform12, internal-plugin, provisioner, puppet]
2019-10-22T01:37:30.305+0300 [DEBUG] plugin: plugin started: path=/usr/bin/terraform12 pid=8093
2019-10-22T01:37:30.305+0300 [DEBUG] plugin: waiting for RPC address: path=/usr/bin/terraform12
2019-10-22T01:37:30.314+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Terraform version: 0.12.10  
2019-10-22T01:37:30.314+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Go runtime version: go1.12.9
2019-10-22T01:37:30.314+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI args: []string{"/usr/bin/terraform12", "internal-plugin", "provisioner", "puppet"}
2019-10-22T01:37:30.314+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] Attempting to open CLI config file: /home/igor_mb/.terraformrc
2019-10-22T01:37:30.314+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019-10-22T01:37:30.314+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI command args: []string{"internal-plugin", "provisioner", "puppet"}
2019-10-22T01:37:30.314+0300 [DEBUG] plugin.terraform12: puppet-provisioner (internal) 2019/10/22 01:37:30 [INFO] Starting provisioner plugin puppet
2019-10-22T01:37:30.315+0300 [DEBUG] plugin.terraform12: plugin address: address=/tmp/plugin225005912 network=unix timestamp=2019-10-22T01:37:30.315+0300
2019-10-22T01:37:30.315+0300 [DEBUG] plugin: using plugin: version=5
2019/10/22 01:37:30 [TRACE] Fetching schema from provisioner "remote-exec"
2019-10-22T01:37:30.316+0300 [DEBUG] plugin: starting plugin: path=/usr/bin/terraform12 args=[/usr/bin/terraform12, internal-plugin, provisioner, remote-exec]
2019-10-22T01:37:30.316+0300 [DEBUG] plugin: plugin started: path=/usr/bin/terraform12 pid=8102
2019-10-22T01:37:30.316+0300 [DEBUG] plugin: waiting for RPC address: path=/usr/bin/terraform12
2019-10-22T01:37:30.326+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Terraform version: 0.12.10  
2019-10-22T01:37:30.326+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Go runtime version: go1.12.9
2019-10-22T01:37:30.326+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI args: []string{"/usr/bin/terraform12", "internal-plugin", "provisioner", "remote-exec"}
2019-10-22T01:37:30.326+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] Attempting to open CLI config file: /home/igor_mb/.terraformrc
2019-10-22T01:37:30.326+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019-10-22T01:37:30.326+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI command args: []string{"internal-plugin", "provisioner", "remote-exec"}
2019-10-22T01:37:30.326+0300 [DEBUG] plugin.terraform12: remote-exec-provisioner (internal) 2019/10/22 01:37:30 [INFO] Starting provisioner plugin remote-exec
2019-10-22T01:37:30.327+0300 [DEBUG] plugin.terraform12: plugin address: address=/tmp/plugin893230264 network=unix timestamp=2019-10-22T01:37:30.327+0300
2019-10-22T01:37:30.327+0300 [DEBUG] plugin: using plugin: version=5
2019/10/22 01:37:30 [TRACE] Fetching schema from provisioner "salt-masterless"
2019-10-22T01:37:30.328+0300 [DEBUG] plugin: starting plugin: path=/usr/bin/terraform12 args=[/usr/bin/terraform12, internal-plugin, provisioner, salt-masterless]
2019-10-22T01:37:30.328+0300 [DEBUG] plugin: plugin started: path=/usr/bin/terraform12 pid=8113
2019-10-22T01:37:30.328+0300 [DEBUG] plugin: waiting for RPC address: path=/usr/bin/terraform12
2019-10-22T01:37:30.338+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Terraform version: 0.12.10  
2019-10-22T01:37:30.338+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] Go runtime version: go1.12.9
2019-10-22T01:37:30.338+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI args: []string{"/usr/bin/terraform12", "internal-plugin", "provisioner", "salt-masterless"}
2019-10-22T01:37:30.338+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] Attempting to open CLI config file: /home/igor_mb/.terraformrc
2019-10-22T01:37:30.338+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2019-10-22T01:37:30.339+0300 [DEBUG] plugin.terraform12: 2019/10/22 01:37:30 [INFO] CLI command args: []string{"internal-plugin", "provisioner", "salt-masterless"}
2019-10-22T01:37:30.339+0300 [DEBUG] plugin.terraform12: salt-masterless-provisioner (internal) 2019/10/22 01:37:30 [INFO] Starting provisioner plugin salt-masterless
2019-10-22T01:37:30.339+0300 [DEBUG] plugin.terraform12: plugin address: address=/tmp/plugin623708899 network=unix timestamp=2019-10-22T01:37:30.339+0300
2019-10-22T01:37:30.339+0300 [DEBUG] plugin: using plugin: version=5
2019/10/22 01:37:30 [TRACE] configupgrade: Working on "dashboard-groups.tf"
2019/10/22 01:37:30 [TRACE] configupgrade: Upgrading signalfx_dashboard_group.dashboardgroup_sda at dashboard-groups.tf:6,1-1
2019/10/22 01:37:30 [TRACE] configupgrade: Upgrading output.dashboardgroup_sda_id at dashboard-groups.tf:11,1-1
2019/10/22 01:37:30 [TRACE] configupgrade: Working on "dashboards-kafka.tf"
2019/10/22 01:37:30 [TRACE] configupgrade: Working on "provider.tf"
2019/10/22 01:37:30 [TRACE] configupgrade: Upgrading provider.signalfx at provider.tf:31,1-1
2019/10/22 01:37:30 [TRACE] configupgrade: Working on "prvider.tf"
2019/10/22 01:37:30 [TRACE] configupgrade: Working on "variables.tf"
2019/10/22 01:37:30 [TRACE] configupgrade: Upgrading var.axis_high_watermark at variables.tf:20,1-1
2019-10-22T01:37:30.342+0300 [DEBUG] plugin: plugin process exited: path=/home/igor_mb/workspace/terraform-sda/signalfx/dashboards/.terraform/plugins/linux_amd64/terraform-provider-signalfx_v4.9.1_x4 pid=8039
2019-10-22T01:37:30.342+0300 [DEBUG] plugin: plugin exited
2019-10-22T01:37:30.343+0300 [DEBUG] plugin: plugin process exited: path=/usr/bin/terraform12 pid=8102
2019-10-22T01:37:30.343+0300 [DEBUG] plugin: plugin exited
2019-10-22T01:37:30.344+0300 [DEBUG] plugin: plugin process exited: path=/usr/bin/terraform12 pid=8083
2019-10-22T01:37:30.344+0300 [DEBUG] plugin: plugin exited
2019-10-22T01:37:30.345+0300 [DEBUG] plugin: plugin process exited: path=/usr/bin/terraform12 pid=8053
2019-10-22T01:37:30.345+0300 [DEBUG] plugin: plugin exited
2019-10-22T01:37:30.346+0300 [DEBUG] plugin: plugin process exited: path=/usr/bin/terraform12 pid=8093
2019-10-22T01:37:30.346+0300 [DEBUG] plugin: plugin exited
2019-10-22T01:37:30.346+0300 [DEBUG] plugin: plugin process exited: path=/usr/bin/terraform12 pid=8113
2019-10-22T01:37:30.346+0300 [DEBUG] plugin: plugin exited
2019-10-22T01:37:30.348+0300 [DEBUG] plugin: plugin process exited: path=/usr/bin/terraform12 pid=8064
2019-10-22T01:37:30.348+0300 [DEBUG] plugin: plugin exited
2019-10-22T01:37:30.348+0300 [DEBUG] plugin: plugin process exited: path=/usr/bin/terraform12 pid=8073
2019-10-22T01:37:30.348+0300 [DEBUG] plugin: plugin exited
panic: strconv.ParseInt: parsing "17976931348623157000": value out of range

goroutine 1 [running]:
github.com/hashicorp/hcl/hcl/token.Token.Value(0x6, 0x0, 0x0, 0x43a, 0x15, 0xd, 0xc000426fe0, 0x14, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/hashicorp/hcl@v0.0.0-20170504190234-a4b07c25de5f/hcl/token/token.go:138 +0x4f8
github.com/hashicorp/terraform/configs/configupgrade.upgradeExpr(0x1ce68e0, 0xc00002d900, 0xc0004401c2, 0xc, 0x0, 0xc0004da240, 0x0, 0x0, 0x0, 0xc0005e0028, ...)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_expr.go:41 +0x3e50
github.com/hashicorp/terraform/configs/configupgrade.upgradeExpr(0x1b62580, 0xc0004a5380, 0xc0004401c2, 0xc, 0x600, 0xc0004da240, 0xc00027aab0, 0x32bf87d, 0xc0005e0578, 0x77aca6, ...)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_expr.go:38 +0x59d8
github.com/hashicorp/terraform/configs/configupgrade.noInterpAttributeRule.func1(0x1b62580, 0xc0004a5380, 0x0, 0x430, 0x15, 0x3, 0xc00034c997, 0x7)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_body.go:43 +0x59
github.com/hashicorp/terraform/configs/configupgrade.attributeRule.func1(0xc00027a8d0, 0xc0004270e0, 0x17, 0xc0004a5320, 0xc0005e1548, 0x1, 0xc00034c997)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_body.go:252 +0x2b6
github.com/hashicorp/terraform/configs/configupgrade.upgradeBlockBody(0xc0004401c2, 0xc, 0xc0004270e0, 0x17, 0xc00027a8d0, 0xc00000ebb0, 0x1, 0x1, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_native.go:469 +0xacc
github.com/hashicorp/terraform/configs/configupgrade.(*Upgrader).upgradeNativeSyntaxFile(0xc0005e1960, 0xc0004401c2, 0xc, 0xc000596000, 0x639, 0x839, 0xc0004da240, 0x1, 0x1, 0x5b9, ...)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade_native.go:166 +0x35db
github.com/hashicorp/terraform/configs/configupgrade.(*Upgrader).Upgrade(0xc0005e1960, 0xc00043e3c0, 0x1e2114d, 0x1, 0xc000448150, 0x3, 0x0, 0x0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/configs/configupgrade/upgrade.go:102 +0x49e
github.com/hashicorp/terraform/command.(*ZeroTwelveUpgradeCommand).Run(0xc0005444e0, 0xc00000c0b0, 0x0, 0x0, 0xc000479030)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/command/012_config_upgrade.go:159 +0x916
github.com/mitchellh/cli.(*CLI).Run(0xc0004297c0, 0xc0004297c0, 0xc00022fd90, 0x1)
	/opt/teamcity-agent/work/9e329aa031982669/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1f1
main.wrappedMain(0x0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:235 +0xb66
main.realMain(0x0)
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:100 +0xb4
main.main()
	/opt/teamcity-agent/work/9e329aa031982669/src/github.com/hashicorp/terraform/main.go:36 +0x3b
```
### Expected Behavior
<!--
What should have happened?
-->

### Actual Behavior
<!--
What actually happened?
-->

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform 0.12upgrade`
-->
1. `terraform init`
2. `terraform 0.12upgrade`
### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- #6017

-->

--- a/configs/configupgrade/upgrade_expr.go
+++ b/configs/configupgrade/upgrade_expr.go
@@ -38,9 +38,9 @@ Value:
 		return upgradeExpr(tv.Token, filename, interp, an)
 
 	case hcl1token.Token:
-		litVal := tv.Value()
 		switch tv.Type {
 		case hcl1token.STRING:
+			litVal := tv.Value()
 			if !interp {
 				// Easy case, then.
 				printQuotedString(&buf, litVal.(string))
@@ -141,6 +141,7 @@ Value:
 			buf.WriteString(marker)
 
 		case hcl1token.BOOL:
+			litVal := tv.Value()
 			if litVal.(bool) {
 				buf.WriteString("true")
 			} else {
@@ -148,12 +149,28 @@ Value:
 			}
 
 		case hcl1token.NUMBER:
-			num := tv.Value()
-			buf.WriteString(strconv.FormatInt(num.(int64), 10))
+			num, err := strconv.ParseInt(tv.Text, 0, 64)
+			if err != nil {
+				diags = diags.Append(&hcl2.Diagnostic{
+					Severity: hcl2.DiagError,
+					Summary:  "Invalid number value",
+					Detail:   fmt.Sprintf("Parsing failed: %s", err),
+					Subject:  hcl1PosRange(filename, tv.Pos).Ptr(),
+				})
+			}
+			buf.WriteString(strconv.FormatInt(num, 10))
 
 		case hcl1token.FLOAT:
-			num := tv.Value()
-			buf.WriteString(strconv.FormatFloat(num.(float64), 'f', -1, 64))
+			num, err := strconv.ParseFloat(tv.Text, 64)
+			if err != nil {
+				diags = diags.Append(&hcl2.Diagnostic{
+					Severity: hcl2.DiagError,
+					Summary:  "Invalid float value",
+					Detail:   fmt.Sprintf("Parsing failed: %s", err),
+					Subject:  hcl1PosRange(filename, tv.Pos).Ptr(),
+				})
+			}
+			buf.WriteString(strconv.FormatFloat(num, 'f', -1, 64))
 
 		default:
 			// For everything else we'll just pass through the given bytes verbatim,

